src/aliceVision/depthMap/cuda/commonStructures.hpp

```diff
@@ -414,27 +499,57 @@ template<class Type, unsigned Dim> void copy(CudaHostMemoryHeap<Type, Dim>& _dst
 {
   cudaMemcpyKind kind = cudaMemcpyDeviceToHost;
   if(Dim == 1) {
-    cudaMemcpy(_dst.getBuffer(), _src.getBuffer(), _src.getBytes(), kind);
+    cudaError_t err = cudaMemcpy(_dst.getBuffer(), _src.getBuffer(), _src.getBytes(), kind);
+    THROW_ON_CUDA_ERROR(err, "Failed to copy (" << __FILE__ << " " << __LINE__ << ")");
   }
   else if(Dim == 2) {
-    cudaMemcpy2D(_dst.getBuffer(), _dst.getSize()[0] * sizeof (Type), _src.getBuffer(), _src.getPitch(), _dst.getSize()[0] * sizeof (Type), _dst.getSize()[1], kind);
+    cudaError_t err = cudaMemcpy2D(_dst.getBuffer(),
+                                   _dst.getSize()[0] * sizeof (Type),
+                                   _src.getBuffer(),
+                                   _src.getPitch(),
+                                   _dst.getSize()[0] * sizeof (Type),
+                                   _dst.getSize()[1], kind);
+    THROW_ON_CUDA_ERROR(err, "Failed to copy (" << __FILE__ << " " << __LINE__ << ")");
   }
   else if(Dim >= 3) {
     for (unsigned int slice=0; slice<_dst.getSize()[2]; slice++)
     {
-      cudaMemcpy2D( _dst.getBuffer() + slice * _dst.getSize()[0] * _dst.getSize()[1], _dst.getSize()[0] * sizeof (Type), &_src.getBuffer()[slice * _src.stride()[1]], _src.getPitch(), _dst.getSize()[0] * sizeof (Type), _dst.getSize()[1], kind);
+      cudaError_t err = cudaMemcpy2D( _dst.getBuffer() + slice * _dst.getSize()[0] * _dst.getSize()[1],
+                                      _dst.getSize()[0] * sizeof (Type),
+                                      (unsigned char*)_src.getBuffer() + slice * _src.stride()[1],
```

Comment:
Line 598 do the same without a cast to `unsigned char`, so one of them is wrong.
I think that stride doesn't include `sizeof(Type)`.